### PR TITLE
Use `low` in `_transform_from_uniform` for TPE sampler

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

See #2391

## Description of the changes

I'm really not familiar with this function, but i think `low` should be added.

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
@nzw0301 Thank you for your PR. If you kindly add a test case to confirm the `low` can be suggested by this change, it will be a great help of the review.

@HideakiImamura Could you review this PR?

@kstoneriv3 If you have time, could you check the change not to break the logic?

**Comment by nzw0301** [comment]
@toshihikoyanase I completely agree with you. I have added the test function for distributions with the `step`-ish arguments to this PR.  Again, I'm afraid that I'm not familiar with this sampler, so I appreciate the knowledgable commuters' comments.

File: tests/samplers_tests/tpe_tests/test_multivariate_parzen_estomator.py [hunk 1]
```diff
@@ -139,6 +139,33 @@ def test_sample_multivariate_parzen_estimator() -> None:
...
+        prior_weight=0.0,
+        consider_magic_clip=False,
+        consider_endpoints=False,
+        weights=lambda x: np.arange(x) + 1.0,
+    )
+
+    # Define search space for distribution with step argument and true ranges
+    search_space = {
+        "c": distributions.DiscreteUniformDistribution(low=1.0, high=7.0, q=3.0),
+        "d": distributions.IntUniformDistribution(1, 5, step=2),
```
Comment (HideakiImamura) [inline tests/samplers_tests/tpe_tests/test_multivariate_parzen_estomator.py:155]: A nit.
```suggestion
        "d": distributions.IntUniformDistribution(low=1, high=5, step=2),
```

**Review by HideakiImamura — APPROVED** [review]
Thanks for the quick fix. Your fix looks perfect. I have a minor comment on the test.

**Comment by kstoneriv3** [comment]
Looks OK to me! Thank you so much for the fix!

**Comment by HideakiImamura** [comment]
@not522 Could you review this PR?

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**